src/components/WorkOrder/Photos/PhotoGroupDescriptionForm.js

```diff
@@ -0,0 +1,106 @@
+import { useForm } from 'react-hook-form'
+import { PrimarySubmitButton, TextArea } from '../../Form'
+import { useState } from 'react'
+
+const PhotoGroupDescriptionForm = ({
+  description,
+  id,
+  onSubmitSetDescription,
+}) => {
+  const [displayForm, setDisplayForm] = useState(false)
+
+  const showForm = () => {
+    setDisplayForm(true)
+  }
+
+  const onSubmit = async (formData) => {
+    const requestBody = {
+      fileGroupId: formData.fileGroupId,
+      description: formData.description,
+    }
+
+    await onSubmitSetDescription(requestBody)
+  }
+
+  if (displayForm) {
+    return (
+      <UpdateDescriptionForm
+        description={description}
+        onSubmit={onSubmit}
+        fileGroupId={id}
+      />
+    )
+  }
+
+  return (
+    <UpdateDescriptionButton description={description} showForm={showForm} />
+  )
+}
+
+export const UpdateDescriptionButton = ({ showForm, description }) => {
```

Comment:
Does this need to be public + tested?
I'd think it could be private and remain just part of the PhotoGroupDescriptionForm component, tested as part of that component also

Reply:
I was actually testing the parent. but i found it was easier to test the children. should probably refactor it